### ISSUE TITLE
Updating lambda to use new image URI fails

### ISSUE THREAD

**Issue by eli88fine** [pulumi/pulumi-aws-native#610]
### What happened?

I tweaked the image URI for a lambda and aws native choked when trying to apply the new change

```
pulumi.automation.errors.CommandError: 
 code: 255
 stdout: Updating (eli):
  pulumi:pulumi:Stack: (same)
    [urn=urn:pulumi:eli::REDACTED::pulumi:pulumi:Stack::REDACTED-eli]
    ~ aws-native:lambda:Function: (update)
        [id=lambda--REDACTED--eli-096d61e]
        [urn=urn:pulumi:eli::REDACTED::aws-native:lambda:Function::lambda--REDACTED--eli]
        [provider: urn:pulumi:eli::REFACTED::pulumi:providers:aws-native::default_0_22_0::9544f70d-099b-47f2-b114-18327fe3ca7f => urn:pulumi:eli::REDACTED::pulumi:providers:aws-native::default_0_26_0::d7b594e5-f261-46e2-af48-755f434d58d6]
      + code: {
          + imageUri: "REDACTED.dkr.ecr.us-east-1.amazonaws.com/REDACTED:event_forwarder-617f03fb8af2a62b1b3d470e7a171350"
        }
Resources:
    27 unchanged

Duration: 4s
 stderr: error: operation error CloudControl: UpdateResource, https response error StatusCode: 400, RequestID: 8724c19b-378c-4e91-9899-d645b57f6393, api error ValidationException: [REPLACE Operation] noSuchPath in source, path provided : //Code
error: update failed
```

### Steps to reproduce

It should be reproducible with any Lambda using an Image as the code source.

```
import json
from typing import Dict
from typing import List
from typing import Optional
from typing import Union

import iam_floyd as statement
from pulumi import export
from pulumi import Output
import pulumi_aws as aws
from pulumi_aws_native import iam
from pulumi_aws_native import lambda_
   


     self.lambda_function = lambda_.Function(
            self.resource_name("lambda"),
            role=self.role.arn,

            code=lambda_.FunctionCodeArgs(image_uri=self._image_uri),
            environment=None
            if self.environment is None
            else lambda_.FunctionEnvironmentArgs(variables=self.environment),
            package_type=lambda_.FunctionPackageType.IMAGE,
            description=self.description,
            timeout=self.timeout,
            reserved_concurrent_executions=self._reserved_concurrent_executions,
        )
```

### Expected Behavior

the lambda would update to use the new Image URI

### Actual Behavior

error

### Output of `pulumi about`

CLI          
Version      3.38.0
Go Version   go1.19
Go Compiler  gc

Host     
OS       debian
Version  10.12
Arch     x86_64

NAME        KIND      VERSION  SIZE    INSTALLED   LAST USED
aws         resource  5.13.0   412 MB  4 days ago  4 days ago
aws-native  resource  0.22.0   84 MB   4 days ago  4 days ago
command     resource  0.4.1    24 MB   4 days ago  4 days ago

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by stack72** [comment]
Hi @eli88fine 

My apologies for this issue - we can get someone to have a look at this ASAP and see what we can do here

Paul

**Comment by eli88fine** [comment]
Great! let me know if there's any additional info I can provide.

**Comment by eli88fine** [comment]
looks like similar behavior is happening with Native S3 Buckets and `pulumi_aws_native.ssm.Document`

**Comment by eli88fine** [comment]
any updates around this?

**Comment by baartch** [comment]
The same happens to me, when updating the content code of a Lambda Function, that was created with *pulumicdk*

```typescript
import * as path from "path";
import * as pulumi from "@pulumi/pulumi";
import * as pulumicdk from "@pulumi/cdk";
import { Duration } from "aws-cdk-lib";
import * as lambda from "aws-cdk-lib/aws-lambda";

class CdkDemoClass extends pulumicdk.Stack {
  fnArn: pulumi.Output<string>;

  constructor(id: string, options?: pulumicdk.StackOptions) {
    super(id, options);

    const fn = new lambda.Function(this, "LambdaFn", {
      code: lambda.Code.fromAsset(path.join(__dirname, "lambda")),
      handler: "index.main",
      timeout: Duration.seconds(29),
      runtime: lambda.Runtime.PYTHON_3_9,
    });

    this.fnArn = this.asOutput(fn.functionArn);

    this.synth();
  }
}

export default CdkDemoClass;

```

I make no changes in the pulumi code. I only change something in the python code of the Lambda, and get the same error:
```bash
     Type                                    Name                                                          Status                  Info
     pulumi:pulumi:Stack                     pulumi-aws-quickstart-dev                                     **failed**              1 error
 ~   ├─ aws:s3:Bucket                        my-bucket                                                     updated (5s)            
     └─ cdk:index:Stack                      CdkDemo                                                                               
 +-     ├─ aws:s3:BucketObjectv2             CdkDemo/CdkDemo/LambdaFn/Code/current_account-current_region  replaced (0.00s)        [diff: ~bucket,source]
        └─ cdk:construct:CdkDemoClass        CdkDemo/CdkDemo                                                                       
           └─ cdk:construct:Function         CdkDemo/CdkDemo/LambdaFn                                                              
 ~            └─ aws-native:lambda:Function  LambdaFn216674B7                                              **updating failed**     [diff: +code]; 1 error


Diagnostics:
  pulumi:pulumi:Stack (pulumi-aws-quickstart-dev):
    error: update failed

  aws-native:lambda:Function (LambdaFn216674B7):
    error: operation error CloudControl: UpdateResource, https response error StatusCode: 400, RequestID: 75941a3f-4785-4c81-aff6-d29a6f0e1940, api error ValidationException: [REPLACE Operation] noSuchPath in source, path provided : //Code
```

**Comment by eli-fine-res** [comment]
Pulumi Team - any updates around this?

**Comment by mikhailshilkov** [comment]
Fixed by #1395
